### PR TITLE
Adding movement to selected device

### PR DESCRIPTION
**Pull request by rodrigogoulartsilva**
Fixing issue #807

### CODE REVIEW

File: simpletransformers/language_representation/representation_model.py [hunk 1]
```diff
@@ -136,10 +136,14 @@ def encode_sentences(self, text_list, combine_strategy=None, batch_size=32):
         :param batch_size
         :return: list of lists of sentence embeddings(if `combine_strategy=None`) OR list of sentence embeddings(if `combine_strategy!=None`)
         """
+
+        self.model.to(self.device)
+
         batches = batch_iterable(text_list, batch_size=batch_size)
         embeddings = np.array([])
         for batch in batches:
```
Comment (aesuli) [inline simpletransformers/language_representation/representation_model.py:144]: I think you should add here:
```
self.model.eval()
with torch.no_grad():
```
to save memory and computation.

Comment (aesuli) [reply simpletransformers/language_representation/representation_model.py:144]: BTW I tested this PR in my code and it works very well. Thanks!

Comment (ThilinaRajapakse) [reply simpletransformers/language_representation/representation_model.py:144]: You are right!

File: simpletransformers/language_representation/representation_model.py [hunk 2]
```diff
@@ -149,13 +153,13 @@ def encode_sentences(self, text_list, combine_strategy=None, batch_size=32):
...
-                batch_embeddings = embedding_func(token_vectors).detach().numpy()
+                batch_embeddings = embedding_func(token_vectors).detach() 
             else:
-                batch_embeddings = token_vectors.detach().numpy()
+                batch_embeddings = token_vectors.detach()
             if len(embeddings) == 0:
-                embeddings = batch_embeddings
+                embeddings = batch_embeddings.cpu().detach().numpy()
             else:
-                embeddings = np.concatenate((embeddings, batch_embeddings), axis=0)
```
Comment (aesuli) [inline simpletransformers/language_representation/representation_model.py:158]: It is more efficient to append the `batch_embeddings` to a list during the loop on batches and then to call the `np.concatenate` only once after the loop ended.

**Review by aesuli — CHANGES_REQUESTED** [review]
Starting from the code in this PR I rewrote the `encode_sentences` in this way (note that it requires the import of tqdm):

```
    def encode_sentences(self, text_list, combine_strategy=None, batch_size=32):
        """
        Generates list of contextual word or sentence embeddings using the model passed to class constructor
        :param text_list: list of text sentences
        :param combine_strategy: strategy for combining word vectors, supported values: None, "mean", "conca\
t"
        :param batch_size
        :return: list of lists of sentence embeddings(if `combine_strategy=None`) OR list of sentence embedd\
ings(if `combine_strategy!=None`)
        """

        self.model.to(self.device)

        batches = batch_iterable(text_list, batch_size=batch_size)
        embeddings = list()
        self.model.eval()
        with torch.no_grad():
            for batch in tqdm(batches, total=np.ceil(len(text_list)/batch_size)):
                input_ids_tensor = self._tokenize(batch)
                input_ids_tensor = input_ids_tensor.to(self.device)
                token_vectors = self.model(input_ids=input_ids_tensor)
                if combine_strategy:
                    embedding_func_mapping = {"mean": mean_across_all_tokens, "concat": concat_all_tokens}
                    if embedding_func_mapping[combine_strategy]:
                        embedding_func = embedding_func_mapping[combine_strategy]
                    else:
                        raise ValueError(
                            "Provided combine_strategy is not valid." "supported values are: 'concat', 'mean\
' and None."
                        )
                    batch_embeddings = embedding_func(token_vectors).detach()
                else:
                    batch_embeddings = token_vectors.detach()
                    embeddings.append(batch_embeddings.cpu().detach().numpy())
        embeddings = np.concatenate(embeddings, axis=0)

        return embeddings

```

**Pull request merged**